### ISSUE TITLE
Lens shows as open and as a process but with no window appearing on screen 

### ISSUE THREAD

**Issue by Ryan-Sharpe1** [lensapp/lens#2195]
**Describe the bug**
No window showing when opening the Lens application. I have tried rebooting my mac, deleting the k8s config folder 
 and reinstalling lens but still no window opening when the process is started. It is showing in the toolbar that Lens is 'open' but with no options on many of the menus as shown in the screenshots attached. My colleagues are not having the same issue and seems to only be on my machine, are there any known fixes? 

**To Reproduce**
1. Unsure we can reproduce this error.
2. Download Lens through github
3. Install the application
4. Open the application
5. No window showing but toolbar shows the programme is 'running' 

**Expected behavior**
For the UI to open. 

**Screenshots**
![Screenshot 2021-02-19 at 12 26 34](https://user-images.githubusercontent.com/54810225/108505510-6be5d600-72af-11eb-9073-0472617a4a74.jpg)
![Screenshot 2021-02-19 at 12 26 55](https://user-images.githubusercontent.com/54810225/108505734-c3844180-72af-11eb-92c8-9a6b613de2bf.jpg)
![Screenshot 2021-02-19 at 12 26 56](https://user-images.githubusercontent.com/54810225/108505773-d26af400-72af-11eb-8bdf-1f72c770ba93.jpg)
![Screenshot 2021-02-19 at 12 26 58](https://user-images.githubusercontent.com/54810225/108505797-deef4c80-72af-11eb-9d8a-46c00668109b.jpg)
![Screenshot 2021-02-19 at 12 26 59](https://user-images.githubusercontent.com/54810225/108505823-e6aef100-72af-11eb-958d-b30280335fb4.jpg)
![Screenshot 2021-02-19 at 12 27 01](https://user-images.githubusercontent.com/54810225/108505854-f0385900-72af-11eb-9452-4ffd02ced60f.jpg)

**Environment (please complete the following information):**
- Lens Version: 4.1.2
- OS: MacOS Catalina 10.15.7 

- Installation method (e.g. snap or AppImage in Linux):
- Lens v4.1.2 - MacOS from github

**Logs:**
When you run the application executable from command line you will see some logging output. Please paste them here:
```
Your logs go here...
```

**Kubeconfig:**
Quite often the problems are caused by malformed kubeconfig which the application tries to load. Please share your kubeconfig, remember to remove any secret and sensitive information.
```
your kubeconfig here
```

**Additional context**

**Comment by arogic** [comment]
Hi, I've got the same issue on Mac BigSur v11.2 with Lens v4.1.2.

**Comment by nevalla** [comment]
Lens tries to sync shell env before opening the app, could it be that some shell script is blocking this execution? For example, we have seen oh-my-zsh blocks execution by prompting to upgrade the tool. If you open new terminal session on your machine, does it open properly?

**Comment by arogic** [comment]
Hi @nevalla, I am using oh-my-zsh, but haven't got any scripts running when I open a terminal session. I tried closing the Terminal and opening a new session, but Lens still doesn't open. However it worked with the previous version of Lens that I had installed. Unfortunately I don't have that version anymore and I don't know the version number.

**Comment by Ryan-Sharpe1** [comment]
I have no shell scripts running on terminal and when I open a new window it opens properly.

**Comment by jim-docker** [comment]
Can you see if there is any log output? 

Lens will store application logs to following locations depending on your operating system: - MacOS: ~/Library/Logs/Lens/ - Windows: %USERPROFILE%\AppData\Roaming\Lens\logs\ - Linux: ~/.config/Lens/logs/

**Comment by arogic** [comment]
Hi @jim-docker, found the logs in `~/Library/Logs/Lens/` and I can see below entries when running `cat Lens.log` for example

`
info: APP:ACTIVATE {"hasVisibleWindows":false}
info: APP:ACTIVATE {"hasVisibleWindows":false}`

Let me know if I should check for something specific in those log files.

Regards

**Comment by Leenie** [comment]
This issue is happening with me too.  I installed the app via github. I then uninstalled it and re-installed via `homebrew`. Same issue. The window doesn't appear. 
Here's the logs from lens:


```➜ ~ tail -f ~/Library/Logs/Lens/lens.log
info: 🚀 Starting Lens from "~/Library/Application Support/Lens"
info: 🐚 Syncing shell environment
info: 💾 Loading stores
info: [STORE]: LOADED from ~/Library/Application Support/Lens/config.json
info: [STORE]: LOADED from ~/Library/Application Support/Lens/lens-cluster-store.json
info: [STORE]: LOADED from ~/Library/Application Support/Lens/lens-workspace-store.json
info: [STORE]: LOADED from ~/Library/Application Support/Lens/lens-extensions.json
info: [STORE]: LOADED from ~/Library/Application Support/Lens/lens-filesystem-provisioner-store.json
info: APP:ACTIVATE {"hasVisibleWindows":false}
```

**Comment by jim-docker** [comment]
To rule out an issue with the existing stores can you try renaming the `~/Library/Application Support/Lens` folder and restarting Lens?

**Comment by nevalla** [comment]
It's a bit surprise to see this line after stores are loaded:
```
info: APP:ACTIVATE {"hasVisibleWindows":false}
```
Maybe user has clicked icon on dock or something that is causing this log line. Or do you see this line always when opening up the app?

Normally after the log line this should come next is:
```
info 🔑 Getting free port for LensProxy server
```
But it's not even go there.

`APP:ACTIVATE` should not do anything at this point since WindowManager is not constructed yet. Or is it? 🤔

**Comment by Leenie** [comment]
@jim-docker I followed your instructions and it's now working, as I can see the window has loaded up on my screen. Thank you so much! 🎉  I've obfuscated the USER ID in the logs: 

```
➜ ~ tail -f ~/Library/Logs/Lens/lens.log
info: [STORE]: LOADED from ~/Library/Application Support/Lens/extension-store/lens-telemetry/preferences-store.json
info: [EXTENSION]: enabled lens-telemetry@0.1.0
info: 📡 Checking for app updates
info: Checking for update
info: Generated new staging user ID: *******************************
error: Can't find working API for the Kubernetes resource ingresses
info: [STORE]: SAVING ~/Library/Application Support/Lens/config.json
info: Update for version 4.1.2 is not available (latest version: 4.1.2, downgrade is disallowed).
info: [STORE]: SAVING ~/Library/Application Support/Lens/lens-workspace-store.json
info: APP:ACTIVATE {"hasVisibleWindows":true}
```

**Comment by Leenie** [comment]
@nevalla I didn't see `info 🔑 Getting free port for LensProxy server` in my logs, but it's now working fine for me. All thanks to @jim-docker 🎉

**Comment by arogic** [comment]
Hi @jim-docker, renaming the `~/Library/Application Support/Lens` Folder helped, but all configuration is lost. Lens is starting now and I can see the window, just need to add all my K8s clusters again. Thanks for your help.

**Comment by jim-docker** [comment]
I have reproduced this so maybe I can figure out the cause

**Comment by rfennellbv** [comment]
> Lens tries to sync shell env before opening the app, could it be that some shell script is blocking this execution? For example, we have seen oh-my-zsh blocks execution by prompting to upgrade the tool. If you open new terminal session on your machine, does it open properly?

This issue had been affecting me for some time and this comment was spot on for me. I had a prompt as part of my `.zshrc`'s first run, which would cause Lens to hang. 
The fix was to open a terminal, deal with the prompt, then kill/relaunch Lens.

**Comment by Ragin-LundF** [comment]
Same here. So far it worked very well but since the update of MacOS 11.2.3 it starts like described behavior here.

I've opened the terminal and started it with

```bash
DEBUG=TRUE /Applications/Lens.app/Contents/MacOS/Lens
```

My logs are:

```
debug: Using host CA: countryName=BM,organizationName=QuoVadis Limited,commonName=QuoVadis Root CA 2
debug: Using host CA: countryName=PL,organizationName=Unizeto Sp. z o.o.,commonName=Certum CA
debug: Using host CA: countryName=US,stateOrProvinceName=New Jersey,localityName=Jersey City,organizationName=The USERTRUST Network,commonName=USERTrust RSA Certification Authority
info: 🚀 Starting Lens from "/Users/ragin/Library/Application Support/Lens"
info: 🐚 Syncing shell environment
```

At this point it hangs.

I've tried already to delete the app, delete the `Library Support/Lens` folder to start it from console or directly...nothing helps.

**Comment by jim-docker** [comment]
@Ragin-LundF your problem looks like it could be a blocked script from the shell environment. Did you try what's suggested in the comment above yours? (https://github.com/lensapp/lens/issues/2195#issuecomment-787916563)
I guess if you've run Lens from the terminal then you have... Is there something in your .bashrc (or .zshrc or ...) that requires user input, which may block Lens's sync with the shell environment?

**Comment by Ragin-LundF** [comment]
@jim-docker : I've also tried to start the app from the terminal to avoid a prompt issue, but it didn't help (that's where I had the logs from).

The prompt can happen on my system too (oh-my-zsh), but it was updated already so I expected no issue here.
I retried to start it from the bash, which has no prompt configured and I also deactivated oh-my-zsh's update prompt:

```bash
# Uncomment the following line to automatically update without prompting.
DISABLE_UPDATE_PROMPT="true"
```

in my .zshrc file without any effect.

Or is there something else that I can do?

**Comment by jim-docker** [comment]
You could make your `.zshrc` as minimal as possible, just for debugging? Other startup scripts you could temporarily disable? From your logs, and the fact that `info: 🐚 Syncing shell environment` is the last log it is likely this is where Lens is hanging.

**Comment by dstotz** [comment]
I had this same exact issue, in my case it turned out to be a fun little Python script I have that runs on shell load https://github.com/thiderman/doge

Disabling that line in my `.zshrc` fixed the issue instantly for me. So you could use that little program to consistently reproduce this issue for debugging purposes.

When I tried launching the app using debug mode I could see it hanging on `Syncing shell environment` and it would sit there forever, but if I did Ctrl+C it would show a shell error and then load the app

 
![image](https://user-images.githubusercontent.com/9381594/114924141-467ad180-9df3-11eb-8847-b6a07ba84d9d.png)

**Comment by giorgian** [comment]
Same problem here: no prompt when launching a new terminal, but replacing `.zshrc` with an empty file works.
Is there a way to detect that zsh is being called by Lens, so that I can make a check at the beginning of my `.zshrc` file and quit immediately if so?

**Comment by dkolb** [comment]
I have this same issue but it appears to be `zinit` causing the issue?  I have a pretty extensive set of zsh plugins that load asynchronously. If I start Lens from an open shell, it launches just fine.  But launching it via Finder or Spotlight breaks just like this issue describes.

Edit 2022 DEC 5: I had (without really thinking it through) exported some environment variables and run `exec zsh` to restart the shell in my `.zshenv` so it would pick up files from the right directory. This apparently breaks any application that tries to utilize a shell in an automated fashion. Removing this exec and rearranging my files to just...work fixed all of it. (Lens, WSL remotes in VSCode, etc.)

**Comment by jim-docker** [comment]
@Ryan-Sharpe1 I think your issue is different than what some others have posted here, and not fixed by #2784. But with a large refactoring happening for Lens 5.0 it may no longer be an issue. If you still encounter the problem with Lens 5.0 when it comes out, please open a new issue about it on github.

**Comment by dmonti** [comment]
Same issue over here on Windows 10, after update to v5.3.2-latest.20211201.1, Lens shows as running on Task Manager but no window appears
